pytest/lib/branches.py

```diff
@@ -4,8 +4,8 @@
 
 
 def current_branch():
-    return subprocess.check_output([
-        "git", "rev-parse", "--abbrev-ref", "HEAD"]).strip().decode()
+    return os.environ.get('BUILDKITE_BRANCH') or subprocess.check_output([
+        "git", "rev-parse", "--symbolic-full-name", "--abbrev-ref", "HEAD"]).strip().decode()
```

Comment:
The first problem on ci fail because this always returns HEAD. so read current branch from env var